### ISSUE TITLE
Illegal memory access in case of corrupted zone.tab

### ISSUE THREAD

**Issue by minichma** [libical/libical#367]
Due to insufficient bound checks in function `fetch_lat_long_from_string()` (icaltimezone.c), parsing of a corrupt `zone.tab` file can lead to illegal memory access. I.e. a trivial file with content like

```
invalid
```

reproduces the problem.

The function first searches to the end of the string [here](https://github.com/libical/libical/blob/5d810d552519b1bb1458e99f043404a80f6c2155/src/libical/icaltimezone.c#L1590). It then increments the pointer [here](https://github.com/libical/libical/blob/5d810d552519b1bb1458e99f043404a80f6c2155/src/libical/icaltimezone.c#L1593), so it points to memory outside the parsed string. However, the pointer is still used in subsequent code, which is illegal. The pattern repeats several time throughout the function.

It's somewhat challenging to provide a reproducing unit test, because
* the code is only used for parsing the system's built-in zone.tab file
* it depends on actual memory content, compiler, etc., whether the problem will cause a crash or be ignored silently

If the app crashes, it's usually due to an assertion in `isspace()` [here](https://github.com/libical/libical/blob/5d810d552519b1bb1458e99f043404a80f6c2155/src/libical/icaltimezone.c#L1606).

A related issue was addressed in #59

**Comment by winterz** [comment]
right. I spent quite some time on fetch_lat_long_from_string() and always ended up breaking things and making them worse.  I'll keep this issue on the pile but without a milestone. maybe someone will have a stroke of genius about it some day.
